src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/WebUI.cs

```diff
@@ -39,45 +39,60 @@ internal abstract class WebUI : IWebUI
     {
         protected Uri RequestUri { get; private set; }
 
-        protected RequestContext context { get; set;}
+        protected RequestContext RequestContext { get; set; }
 
         protected Uri CallbackUri { get; private set; }
 
-        public Object OwnerWindow { get; set; }
+        public object OwnerWindow { get; set; }
+        protected SynchronizationContext SynchronizationContext { get; set; }
 
         public async Task<AuthorizationResult> AcquireAuthorizationAsync(Uri authorizationUri, Uri redirectUri, RequestContext requestContext)
         {
             AuthorizationResult authorizationResult = null;
 
-            var sendAuthorizeRequest = new Action(
-                delegate
-                {
-                    authorizationResult = this.Authenticate(authorizationUri, redirectUri);
-                });
+            var sendAuthorizeRequest = new Action(() =>
```

Comment:
Could you create these variables inside the if / else blocks? It might be easier to read the code.